application/frontend/controllers/MfaController.php

```diff
@@ -95,18 +129,53 @@ public function actionVerify($mfaId)
                 \Yii::$app->response->statusCode = 204;
                 return;
             }
-        } catch (\Exception $e) {
+        } catch (ServiceException $e) {
             \Yii::error([
-                'action' => 'verify mfa',
-                'status' => 'error',
+                'status' => 'MFA verify error',
                 'error' => $e->getMessage(),
-                'code' => $e->getCode(),
-                'user' => \Yii::$app->user->identity->email,
-            ]);
-            throw new ServerErrorHttpException('Unable to verify MFA code, error code: ' . $e->getCode());
+            ], __METHOD__);
+            if ($e->httpStatusCode == 404) {
+                throw new NotFoundHttpException(\Yii::t('app', 'MFA verify failure'), $e->getCode());
+            }
+
+            /*
+             * Other status codes will result in a 500 response
+             */
+            throw $e;
+        } catch (MfaRateLimitException $e) {
+            throw new TooManyRequestsHttpException(\Yii::t('app', 'MFA rate limit failure'), $e->getCode());
         }
 
         throw new BadRequestHttpException(\Yii::t('app', 'Invalid code provided'));
     }
 
+    /**
+     * @param $mfaId
+     * @throws NotFoundHttpException
+     * @throws ServiceException
+     */
+    public function actionUpdate($mfaId)
+    {
+        $label = \Yii::$app->request->getBodyParam('label');
+        if ($label === null) {
+            return;
```

Comment:
should this throw some kind of `400`?

Reply:
I didn't think so. With a high-level view of `PUT`, it only changes properties that are present. Do you think it should be required?

Reply:
no, I don't think it needs to be required.  I'm just questioning the `return`.  What does the response look like when `label = null`?